openscap/openscap.go

```diff
@@ -0,0 +1,106 @@
+package openscap
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+)
+
+const (
+	CPE               = "oval:org.open-scap.cpe.rhel:def:"
+	MAX_DIST          = 7
+	MIN_DIST          = 5
+	CPE_DICT          = "/usr/share/openscap/cpe/openscap-cpe-oval.xml"
+	CVE_URL           = "https://www.redhat.com/security/data/metrics/ds/"
+	DIST_CVE_NAME_FMT = "com.redhat.rhsa-RHEL%d.ds.xml.bz2"
+)
+
+func getRHELDist(chrootPath string) (int, error) {
+	for dist := MIN_DIST; dist <= MAX_DIST; dist = dist + 1 {
+		output, err := oscapChroot(chrootPath, "oval", "eval", "--id",
+			fmt.Sprintf("%s%d", CPE, dist), CPE_DICT, "2>&1", ">", "/dev/null")
+		if err != nil {
+			return 0, err
+		}
+		if strings.Contains(string(output), fmt.Sprintf("%s%d: true", CPE, dist)) {
+			return dist, nil
+		}
+	}
+
+	return 0, fmt.Errorf("could not find RHEL dist")
+}
+
+func getInputCVE(cveDir string, dist int) (string, error) {
+	cveName := fmt.Sprintf(DIST_CVE_NAME_FMT, dist)
+	cveFileName := path.Join(cveDir, cveName)
+	cveURL, _ := url.Parse(CVE_URL)
+	cveURL.Path = path.Join(cveURL.Path, cveName)
+
+	out, err := os.Create(cveFileName)
+	if err != nil {
+		return "", fmt.Errorf("Could not create file %s: %v\n", cveFileName, err)
+	}
+	defer out.Close()
+
+	resp, err := http.Get(cveURL.String())
+	if err != nil {
+		return "", fmt.Errorf("Could not download file %s: %v\n", cveURL, err)
+	}
+	defer resp.Body.Close()
+
+	_, err = io.Copy(out, resp.Body)
+	return cveFileName, err
+}
+
+func setOscapChrootEnv(chrootPath string) error {
+	os.Setenv("OSCAP_PROBE_ROOT", chrootPath)
+
+	out, err := exec.Command("uname", "-a").Output()
+	if err != nil {
+		return fmt.Errorf("could not run uname -a %v", err)
+	}
+	info := strings.Split(string(out), " ")
+	os.Setenv("OSCAP_PROBE_ARCHITECTURE", info[len(info)-1])
+	os.Setenv("OSCAP_PROBE_OS_NAME", info[0])
+	os.Setenv("OSCAP_PROBE_OS_VERSION", info[2])
+	os.Setenv("OSCAP_PROBE_PRIMARY_HOST_NAME", "inspected_image")
```

Comment:
These values go down to final report of the scan. That also means that they are shown to user in final HTML report.

We don't process these items in any way, unless someone writes a security policy that assert for them.